Reinstated the `@metadata` decorator using environment variables inst…

frankcorneliusmartin opened the pull request:
…ead of JWT token extraction

bartvanb reviewed (changes requested):
OK fine by me. I think it would be important to test that this really works, maybe we can add that to #2213? And can you also add that the docs need to be written to one of the docs issues (or create a new one)? Otherwise we would probably forget

frankcorneliusmartin commented:
No worries. I added a paragraph of documentation about this decorator and added to the #2213 that we need to verify that the decorator actually works. Let me know if we can merge it like this.

frankcorneliusmartin commented:
I now also tested it using the (new) mock client. Ill leave the remark in #2213 to check it in a real network.
